### PR TITLE
EASY-1902 : flatten AccessRights.category (breaks ui and pending drafts)

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes EASY-1902 : flatten AccessRights.category

#### When applied it will
* reduce a complex json field to a simple one:
  old: ` "accessRights": { "category": "OPEN_ACCESS" },`
  new: `"accessRights": "OPEN_ACCESS",`
* 
* 

#### How should this be manually tested?

* [x] avoid class cast
* [x] more unit tests

#### Related pull requests on github
* [x] easy-deposit-ui -> https://github.com/DANS-KNAW/easy-deposit-ui/pull/137

### CODE REVIEW

File: src/test/scala/nl.knaw.dans.easy.deposit/servlets/DoiSpec.scala [hunk 1]
```diff
@@ -36,7 +36,7 @@ class DoiSpec extends DepositServletFixture {
        |    { "scheme": "dcterms:W3CDTF", "value": "2018-05-31", "qualifier": "dcterms:created" },
        |  ],
        |  "audiences": [ { "scheme": "string", "value": "string", "key": "D33000" } ],
-       |  "accessRights": {"category": "OPEN_ACCESS"},
+       |  "accessRights": "OPEN_ACCESS",
```
Comment (jo-pol) [inline src/test/scala/nl.knaw.dans.easy.deposit/servlets/DoiSpec.scala]: Which name should we use, `accessRights`, `accessCategory` or something else?

Comment (rvanheest) [reply src/test/scala/nl.knaw.dans.easy.deposit/servlets/DoiSpec.scala]: @lindareijnhoudt

Comment (lindareijnhoudt) [reply src/test/scala/nl.knaw.dans.easy.deposit/servlets/DoiSpec.scala]: i'm fine with accessRights

File: src/test/scala/nl.knaw.dans.easy.deposit/docs/JsonUtilSpec.scala [hunk 2]
```diff
@@ -53,4 +55,66 @@ class JsonUtilSpec extends TestSupportFixture {
...
+    }
+  }
+
+  it should "extract just the last object" in {
+    deserializeAllOptions(
+      """{"x": "foo bar"}
+        |{"xs": [ "123abc" ]
+        |}""".stripMargin
+    ) should matchPattern {
+      case Success(AllOptions(None, Some(_))) =>
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/docs/JsonUtilSpec.scala]: add more detail to the pattern match

File: src/test/scala/nl.knaw.dans.easy.deposit/docs/JsonUtilSpec.scala [hunk 3]
```diff
@@ -53,4 +55,66 @@ class JsonUtilSpec extends TestSupportFixture {
...
+    }
+  }
+
+  it should "reject an empty array" in {
+    deserializeAllOptions("""[]""") should matchPattern {
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/docs/JsonUtilSpec.scala]: in a number of cases you don't need the triple-quotes

**Pull request merged**